### DISCUSSION TITLE
UI Dashboard: Database Functions: can't edit, arguments get wiped

### DISCUSSION THREAD

**Discussion by stuartcrobinson** [supabase/supabase#16936] [Contribute to Supabase]
can't edit or display database function arguments from the UI, especially functions created from the UI

# Steps to reproduce:

creating a function:
<img width="787" alt="Screen Shot 2023-01-17 at 1 40 00 PM" src="https://user-images.githubusercontent.com/12928716/212983978-e790290a-1454-4256-b141-84373a5fd83b.png">
now trying to edit that same function:
<img width="801" alt="Screen Shot 2023-01-17 at 1 40 17 PM" src="https://user-images.githubusercontent.com/12928716/212983975-cc022ea1-e635-4704-bca3-1e30a0ed72d6.png">
can't see or edit arguments:
<img width="792" alt="Screen Shot 2023-01-17 at 1 40 31 PM" src="https://user-images.githubusercontent.com/12928716/212983974-59c1294a-adad-48db-97f9-7b82e0bf04d7.png">

### note: it displays arguments for functions defined in the project .sql file, but still wont let me edit them:
<img width="678" alt="image" src="https://user-images.githubusercontent.com/12928716/212985832-5073d365-dba9-41ec-a064-6702ca74ba46.png">

**Comment by GaryAustin1** [comment]
Postgres considers the parameter names and types as a fixed part of the function.  If they change you have a new function and the old one still exists with the other parameters.

https://www.postgresql.org/docs/current/sql-createfunction.html#SQL-CREATEFUNCTION-DESCRIPTION

`To replace the current definition of an existing function, use CREATE OR REPLACE FUNCTION. It is not possible to change the name or argument types of a function this way (if you tried, you would actually be creating a new, distinct function). Also, CREATE OR REPLACE FUNCTION will not let you change the return type of an existing function. To do that, you must drop and recreate the function. (When using OUT parameters, that means you cannot change the types of any OUT parameters except by dropping the function.)`

So normally you have to delete the function and create a new one even in SQL.

**Comment by stuartcrobinson** [comment]
^^ interesting

coming back to add that then it wouldn't let me edit the function at all. that "Name of argument" is not editable

![image](https://user-images.githubusercontent.com/12928716/212991649-39abe8c8-1ac1-4326-80ac-b05f7635c9f2.png)
